test/models/setting_test.rb

```diff
@@ -6,6 +6,11 @@ class SettingTest < ActiveSupport::TestCase
   should validate_uniqueness_of(:name)
   should validate_inclusion_of(:settings_type).in_array(Setting::TYPES)
 
+  setup do
+    # Activate caching for this testsuite
+    Setting.stubs(:cache).returns(Rails.cache)
+  end
```

Comment:
This guess we should unstub this in `teardown` or?

Reply:
I don't think, this is required. Mocha actually cleans up after each test, see https://github.com/freerange/mocha/blob/065cacd6e6b87d9852a6740393d3cdba3570773f/lib/mocha/integration/mini_test/version_230_to_2101.rb#L56-L57

Reply:
Oh. So the issue wasn't that other tests would be affected by this, bur rather the facts ignoring test failing when some plugin was around?

Reply:
Not exactly, let's take a step back.

Plugin test also run the whole core test suite. When a plugin extends the Host model (e.g. by a before_validate hook or something) and accesses the `Setting` model there, that call is not stubbed in the core test suite. The test in the core test suite fails then.

Real world example:
https://github.com/theforeman/foreman_expire_hosts/blob/ffbe87e35c1deadb973924ca56cce9b25f7852b4/app/models/concerns/foreman_expire_hosts/host_ext.rb#L8

Another (unreal) approach how to fix this would be to extend the core test from plugins (that's not possible to my knowledge) like this:

```
Setting.stubs(:[]).with(:discovery_reboot).returns('whatever')
Setting.stubs(:[]).with(:is_host_expiry_date_mandatory).returns('whatever')
```

You can see an example (discovery) here:

http://ci.theforeman.org/job/test_plugin_pull_request/4468/

I hope, the motivation for this is clear now. I'll set this for 1.17 as this affects two plugins.

Reply:
Everything clear now, works for me. Merging now